include/nanobind/eigen/dense.h

```diff
@@ -28,6 +28,28 @@ NAMESPACE_BEGIN(detail)
 template <typename T>
 constexpr int NumDimensions = bool(T::IsVectorAtCompileTime) ? 1 : 2;
 
+template<typename T>
+struct StrideExtr
+{
+    using Type = Eigen::Stride<0, 0>;
+};
+
+template <typename T, int Options, typename StrideType>
+struct StrideExtr<Eigen::Map<T, Options, StrideType>>
+{
+    using Type = StrideType;
+};
+
+template<typename T>
+using Stride = typename StrideExtr<T>::Type;
+
+template<typename T>
```

Comment:
Could you add a comment explaining the different cases being considered here.  I looked at it and couldn't understand the whole thing.

Please rename top-level constexpr to snake_case (`requires_contig_memory`)